Skeletal Resource: AWS Route Table

miah opened the pull request:
This provides a skeletal AWS Route Table resource for InSpec.

https://github.com/chef/inspec-aws/issues/168

miah committed:
Create a AWS Route Table Resource.

Signed-off-by: Miah Johnson <miah@chia-pet.org>

clintoncwolfe commented on libraries/aws_route_table.rb:
```diff
@@ -0,0 +1,69 @@
+      raise ArgumentError, 'aws_route_table Route Table ID must be in the' \
+       ' format "rtb-" followed by 8 hexadecimal characters.'
+    end
+
```
Is it OK to not pass the route table ID?  What is the behavior?

clintoncwolfe commented on libraries/aws_route_table.rb:
```diff
@@ -0,0 +1,69 @@
+    @routetable = resp.to_h[:route_tables]
+
+    unless @routetable.empty?
+      r = @routetable.first
```
So, if you pass a route table ID, you get that one, but if you leave it out, you get a random route table?

clintoncwolfe commented on libraries/aws_route_table.rb:
```diff
@@ -0,0 +1,69 @@
+    end
+
+    resp = backend.describe_route_tables(args)
+    @routetable = resp.to_h[:route_tables]
```
`@routetable` should probably be a local variable, not an instance variable.

clintoncwolfe commented on test/integration/default/build/route_table.tf:
```diff
@@ -0,0 +1,48 @@
+    gateway_id = "${data.aws_internet_gateway.default.id}"
+  }
+
+  tags {
```
Minor thing, but since we're not testing for tags yet, and we have a dedicated test AWS account, I generally omit Tags.  The idea being that the things in the test fixture file are the minimal number of things we need to test, and each thing is important.

miah replied:
Ok I can remove this for now.

clintoncwolfe commented on test/integration/default/verify/controls/aws_route_table.rb:
```diff
@@ -0,0 +1,39 @@
+  fixtures[fixture_name] = attribute(
+    fixture_name,
+    default: "default.#{fixture_name}",
+    description: 'See ../build/ec2.tf',
```
route_table.tf, actually

clintoncwolfe commented on test/integration/default/verify/controls/aws_route_table.rb:
```diff
@@ -0,0 +1,39 @@
+  end
+end
+
+control "aws_vpc properties" do
```
aws_route_table properties - aws_vpc would make for some confusing test output!

miah replied:
Doh =)

clintoncwolfe commented on test/integration/default/verify/controls/aws_route_table.rb:
```diff
@@ -0,0 +1,39 @@
+  describe aws_route_table(fixtures['routetable_rtb_route_table_id']) do
+    its('vpc_id') { should eq fixtures['routetable_rtb_vpc_id'] }
+    its('tags') { should eq fixtures['routetable_rtb_tags'] }
+    its('routes') { should eq fixtures['routetable_rtb_routes']}
```
Huh.  Does our "TF output => YAML => attribute loader => fixture hash" system preserve arrays / deep data structures?

miah replied:
When I threw a puts in my control I saw a hash, I'll triple check this though.

clintoncwolfe reviewed (dismissed):
Great work!
Biggest thing: missing the doc file!
Beyond that, just a few minor suggestions.  Tests pass fine.  Thanks!

miah committed:
Change instance variable to local.
Create documentation.
Remove unsupported properties.
Clean up tests to support single argument and hash.

Signed-off-by: Miah Johnson <miah@chia-pet.org>

miah commented:
I cleaned this up and paired it down to the basics. Thanks for the comments @clintoncwolfe =)

miah committed:
Fix doc and remove last property.
If you don't supply any values, and the response isn't empty, a route table exists.
Remove tests for properties.

Signed-off-by: Miah Johnson <miah@chia-pet.org>

clintoncwolfe reviewed (approved):
Looks great - thanks for the contribution and great docs!
